examples/lighting-app/nrfconnect/main/AppTask.cpp

```diff
@@ -307,18 +309,47 @@ void AppTask::FunctionHandler(AppEvent * aEvent)
     }
 }
 
+void AppTask::StartThread()
+{
+    chip::DeviceLayer::Internal::DeviceNetworkInfo deviceNetworkInfo;
+    memset(&deviceNetworkInfo, 0, sizeof(deviceNetworkInfo));
+
+    const char* networkName = "OpenThreadDemo";
+    const uint64_t networkExtendedPANID = 0x2222222211111111;
+    const uint8_t masterKey[chip::DeviceLayer::Internal::kThreadNetworkKeyLength] = {0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,\
+                                                                                     0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+    const uint8_t threadMeshPrefix[chip::DeviceLayer::Internal::kThreadMeshPrefixLength] = {0xfd, 0x11, 0x11, 0x11, 0x11, 0x22, 0x00, 0x00};
+
+    memcpy(deviceNetworkInfo.ThreadNetworkName, networkName, strlen(networkName));
+    memcpy(deviceNetworkInfo.ThreadExtendedPANId, &networkExtendedPANID, sizeof(networkExtendedPANID));
+    deviceNetworkInfo.FieldPresent.ThreadExtendedPANId = true;
+    memcpy(deviceNetworkInfo.ThreadNetworkKey, masterKey, sizeof(masterKey));
+    deviceNetworkInfo.FieldPresent.ThreadMeshPrefix = true;
+    memcpy(deviceNetworkInfo.ThreadMeshPrefix, threadMeshPrefix, sizeof(threadMeshPrefix));
+    deviceNetworkInfo.ThreadPANId = 0x1234;
+    deviceNetworkInfo.ThreadChannel = 15;
+
+    if (sIsThreadEnabled) {
+        ThreadStackMgr().SetThreadEnabled(false);
+    }
+
+    ThreadStackMgr().SetThreadProvision(deviceNetworkInfo);
+    ThreadStackMgr().SetThreadEnabled(true);
+}
+
 void AppTask::JoinerHandler(AppEvent * aEvent)
 {
     if (aEvent->ButtonEvent.PinNo != JOINER_BUTTON)
         return;
 
-    CHIP_ERROR error = CHIP_ERROR_NOT_IMPLEMENTED;
-
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    error = ThreadStackMgr().JoinerStart();
+    if (!otDatasetIsCommissioned(ThreadStackMgrImpl().OTInstance())) {
```

Comment:
Should we use OpenThread calls here directly or crate a dedicate method inside `ThreadStackMgrImpl()`?

Reply:
I found dedicated method inside of the ConnectivityMgr. Is it better now?